packages/suite/src/actions/firmware/firmwareActions.ts

```diff
@@ -55,21 +84,24 @@ export const setTargetRelease = (payload: AcquiredDevice['firmwareRelease']): Fi
 const firmwareInstall =
     (fwBinary?: ArrayBuffer, firmwareType?: FirmwareType) =>
     async (dispatch: Dispatch, getState: GetState) => {
-        const { device } = getState().suite;
-        const { targetRelease, prevDevice, useDevkit, intermediaryInstalled } = getState().firmware;
+        const device = selectDevice(getState());
+        const targetRelease = selectTargetRelease(getState());
+        const prevDevice = selectPrevDevice(getState());
+        const useDevkit = selectUseDevkit(getState());
+        const intermediaryInstalled = selectIntermediaryInstalled(getState());
```

Comment:
nit: I guess here using selectors does not benefit anything, since the majority is just different properties of `firmware`